setup.py

```diff
@@ -326,6 +327,11 @@ def awscrt_ext():
     libraries.reverse()
 
     if sys.platform == 'win32':
+        # distutils is deprecated in Python 3.10 and removed in 3.12. However, it still works because Python defines a compatibility interface as long as setuptools is installed.
+        # We don't have an official alternative for distutils.ccompiler as of September 2024. See: https://github.com/pypa/setuptools/issues/2806
+        # Once that issue is resolved, we can migrate to the official solution.
+        # For now, restrict distutils to Windows only, where it's needed.
+        import distutils.ccompiler
```

Comment:
Can we move the import to the beginning of the file instead? So that we don't need to have two place to import the package?

Reply:
If we move the import to the beginning of the file, it will be imported everywhere. Because this is only required on Windows and is deprecated, I have restricted it to Windows.

Reply:
I mean move the
```
if sys.platform == 'win32':
   import distutils.ccompiler
```
to the top of the file

Reply:
Sorry, missed it in this PR. I have created #599 to fix this.